jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java

```diff
@@ -0,0 +1,184 @@
+import javax.management.openmbean.OpenType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.lafaspot.jmetrics.annotation.MetricClass;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfoList;
+import io.github.classgraph.ScanResult;
+
+/**
+ * Used by public to get all monitor metricClass name and method metric name, type.
+ *
+ * @author KevinL
+ *
+ */
+public class MonitorScanner implements MonitorScannerMBean {
+    /** Get logger. */
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    /** Monitor compositeData send to yamas. */
+    private CompositeData allMonitorCompositeData;
+    /** Whitelist package to scan by classgraph. */
+    private final List<String> whitelistedPackagesList;
+    /** Blacklist package to scan by classgraph. */
+    private final List<String> blacklistedPackagesList;
+
+    /**
+     * Construtor. Scan all monitor classes.
+     * 
+     * @param whitelistedPackagesList Whitelist packages to scan. Empty for whitelist all.
+     * @param blacklistedPackagesList Blacklist packages to scan. Empty for empty blacklist.
+     */
+    public MonitorScanner(@Nonnull final List<String> whitelistedPackagesList, @Nonnull final List<String> blacklistedPackagesList) {
+        allMonitorCompositeData = null;
+        this.whitelistedPackagesList = whitelistedPackagesList;
+        this.blacklistedPackagesList = blacklistedPackagesList;
+    }
+
+    /**
+     * Get All monitors and store metric metricClass to compositeData[].
+     *
+     * @throws OpenDataException creating compositetype fail
+     */
+    private void scan() {
+        final Set<Class<?>> monitors = new HashSet<>();
+        final ClassGraph classGraph = new ClassGraph();
+        classGraph.enableAnnotationInfo().ignoreClassVisibility().blacklistLibOrExtJars()
+                .removeTemporaryFilesAfterScan()
+                .overrideClassLoaders(this.getClass().getClassLoader()).ignoreParentClassLoaders();
+        if (whitelistedPackagesList.size() > 0) {
+            classGraph.whitelistPackages(whitelistedPackagesList.toString());
+        }
+        if (blacklistedPackagesList.size() > 0) {
+            classGraph.blacklistPackages(blacklistedPackagesList.toString());
+        }
+        final ScanResult scanResult = classGraph.scan();
+        final ClassInfoList classInfoList = scanResult.getClassesWithAnnotation(MetricClass.class.getName());
+        monitors.addAll(classInfoList.loadClasses(true));
+
+        final CompositeData[] monitorCompositeData = new CompositeData[monitors.size()];
+            int i = 0;
+        for (Class<?> monitor : monitors) {
+            try {
+                monitorCompositeData[i] = new MonitorCompositeData(monitor).getMetricAnnotation();
+                i++;
+            } catch (OpenDataException ox) {
+                logger.error("Getting monitor metricaAnnotation failed: " + monitor.getName());
```

Comment:
You will have empty monitorCompositeData when this happens. Your subsequent code has to deal with that.